File: 2021/go/day01/day01_test.go
package main

import (
	"fmt"
	"testing"

	"github.com/charlesbourget/aoc-lib/lib"
)

func TestPart1(t *testing.T) {
	input, err := lib.ReadInt("../../inputs/day01/input.test")
	if err != nil {
		fmt.Println("Error while reading input. ", err)
		return
	}

	expected := 7
	result := Part1(input)
	if result != expected {
		t.Fatalf(`Part1() = %d, want %d, error`, result, expected)
	}
}

func TestPart2(t *testing.T) {
	input, err := lib.ReadInt("input.test")
	if err != nil {
		fmt.Println("Error while reading input. ", err)
		return
	}

	expected := 5
	result := Part2(input)
	if result != expected {
		t.Fatalf(`Part2() = %d, want %d, error`, result, expected)
	}
}

File: 2023/haskell-aoc/tests/Day01Test.hs
module Day01Test (tests) where

import Day01 (part1, part2)
import Input (readInput)
import Test.HUnit

tests :: Test
tests = TestList [testPart1]

testPart1 :: Test
testPart1 =
  TestCase
    ( do
        input <- readInput "../inputs/day01/input.test"
        assertEqual "Day 01, Part 1" (part1 input) 142
    )

testPart2 :: Test
testPart2 =
  TestCase
    ( do
        input <- readInput "../inputs/day01/input.test"
        assertEqual "Day 01, Part 2" (part2 input) 281
    )
